### PR TITLE
Cleanup tests

### PR DESCRIPTION
**Pull request by ChrisThrasher**
## Description

Here are some assorted cleanups to the tests I've been accumulated for a couple months. No drastic changes are made. The biggest change is renaming all the tests cases. Here's a before and after for what the test output looks like. I think it's a lot easier to parse now.

Before
```
Run tests
Test project /Users/thrasher/Projects/sfml/build/test
      Start  1: sf::Angle class - [system]
 1/28 Test  #1: sf::Angle class - [system] ...............   Passed    0.00 sec
      Start  2: sf::Clock class - [system]
 2/28 Test  #2: sf::Clock class - [system] ...............   Passed    0.01 sec
      Start  3: SFML/Config.hpp
 3/28 Test  #3: SFML/Config.hpp ..........................   Passed    0.00 sec
      Start  4: sf::err - [system]
 4/28 Test  #4: sf::err - [system] .......................   Passed    0.00 sec
      Start  5: sf::FileInputStream class - [system]
 5/28 Test  #5: sf::FileInputStream class - [system] .....   Passed    0.00 sec
      Start  6: sf::MemoryInputStream class - [system]
 6/28 Test  #6: sf::MemoryInputStream class - [system] ...   Passed    0.00 sec
      Start  7: sf::Time class - [system]
 7/28 Test  #7: sf::Time class - [system] ................   Passed    0.00 sec
      Start  8: sf::Vector2 class template - [system]
 8/28 Test  #8: sf::Vector2 class template - [system] ....   Passed    0.00 sec
      Start  9: sf::Vector3 class template - [system]
 9/28 Test  #9: sf::Vector3 class template - [system] ....   Passed    0.00 sec
      Start 10: sf::ContextSettings class - [window]
10/28 Test #10: sf::ContextSettings class - [window] .....   Passed    0.01 sec
      Start 11: sf::VideoMode class - [window]
11/28 Test #11: sf::VideoMode class - [window] ...........   Passed    0.01 sec
      Start 12: sf::BlendMode class - [graphics]
12/28 Test #12: sf::BlendMode class - [graphics] .........   Passed    0.01 sec
      Start 13: sf::CircleShape class - [graphics]
13/28 Test #13: sf::CircleShape class - [graphics] .......   Passed    0.01 sec
      Start 14: sf::Color class - [graphics]
14/28 Test #14: sf::Color class - [graphics] .............   Passed    0.01 sec
      Start 15: sf::ConvexShape class - [graphics]
15/28 Test #15: sf::ConvexShape class - [graphics] .......   Passed    0.01 sec
      Start 16: sf::Glyph class - [graphics]
16/28 Test #16: sf::Glyph class - [graphics] .............   Passed    0.01 sec
      Start 17: sf::Image - [graphics]
17/28 Test #17: sf::Image - [graphics] ...................   Passed    0.01 sec
      Start 18: sf::Rect class template - [graphics]
18/28 Test #18: sf::Rect class template - [graphics] .....   Passed    0.01 sec
      Start 19: sf::RectangleShape class - [graphics]
19/28 Test #19: sf::RectangleShape class - [graphics] ....   Passed    0.01 sec
      Start 20: sf::RenderStates class - [graphics]
20/28 Test #20: sf::RenderStates class - [graphics] ......   Passed    0.01 sec
      Start 21: sf::Shape class - [graphics]
21/28 Test #21: sf::Shape class - [graphics] .............   Passed    0.01 sec
      Start 22: sf::Transform class - [graphics]
22/28 Test #22: sf::Transform class - [graphics] .........   Passed    0.01 sec
      Start 23: sf::Transformable class - [graphics]
23/28 Test #23: sf::Transformable class - [graphics] .....   Passed    0.01 sec
      Start 24: sf::Vertex class - [graphics]
24/28 Test #24: sf::Vertex class - [graphics] ............   Passed    0.01 sec
      Start 25: sf::VertexArray class - [graphics]
25/28 Test #25: sf::VertexArray class - [graphics] .......   Passed    0.01 sec
      Start 26: sf::View class - [graphics]
26/28 Test #26: sf::View class - [graphics] ..............   Passed    0.01 sec
      Start 27: sf::IpAddress class - [network]
27/28 Test #27: sf::IpAddress class - [network] ..........   Passed    0.28 sec
      Start 28: sf::Packet class - [network]
28/28 Test #28: sf::Packet class - [network] .............   Passed    0.00 sec
```

After
```
Run tests
Test project /Users/thrasher/Projects/sfml/build/test
      Start  1: [System] sf::Angle
 1/28 Test  #1: [System] sf::Angle ...............   Passed    0.00 sec
      Start  2: [System] sf::Clock
 2/28 Test  #2: [System] sf::Clock ...............   Passed    0.01 sec
      Start  3: SFML/Config.hpp
 3/28 Test  #3: SFML/Config.hpp ..................   Passed    0.00 sec
      Start  4: [System] sf::err
 4/28 Test  #4: [System] sf::err .................   Passed    0.00 sec
      Start  5: [System] sf::FileInputStream
 5/28 Test  #5: [System] sf::FileInputStream .....   Passed    0.00 sec
      Start  6: [System] sf::MemoryInputStream
 6/28 Test  #6: [System] sf::MemoryInputStream ...   Passed    0.00 sec
      Start  7: [System] sf::Time
 7/28 Test  #7: [System] sf::Time ................   Passed    0.00 sec
      Start  8: [System] sf::Vector2
 8/28 Test  #8: [System] sf::Vector2 .............   Passed    0.00 sec
      Start  9: [System] sf::Vector3
 9/28 Test  #9: [System] sf::Vector3 .............   Passed    0.00 sec
      Start 10: [Window] sf::ContextSettings
10/28 Test #10: [Window] sf::ContextSettings .....   Passed    0.01 sec
      Start 11: [Window] sf::VideoMode
11/28 Test #11: [Window] sf::VideoMode ...........   Passed    0.01 sec
      Start 12: [Graphics] sf::BlendMode
12/28 Test #12: [Graphics] sf::BlendMode .........   Passed    0.01 sec
      Start 13: [Graphics] sf::CircleShape
13/28 Test #13: [Graphics] sf::CircleShape .......   Passed    0.01 sec
      Start 14: [Graphics] sf::Color
14/28 Test #14: [Graphics] sf::Color .............   Passed    0.01 sec
      Start 15: [Graphics] sf::ConvexShape
15/28 Test #15: [Graphics] sf::ConvexShape .......   Passed    0.01 sec
      Start 16: [Graphics] sf::Glyph
16/28 Test #16: [Graphics] sf::Glyph .............   Passed    0.01 sec
      Start 17: [Graphics] sf::Image
17/28 Test #17: [Graphics] sf::Image .............   Passed    0.01 sec
      Start 18: [Graphics] sf::Rect
18/28 Test #18: [Graphics] sf::Rect ..............   Passed    0.01 sec
      Start 19: [Graphics] sf::RectangleShape
19/28 Test #19: [Graphics] sf::RectangleShape ....   Passed    0.01 sec
      Start 20: [Graphics] sf::RenderStates
20/28 Test #20: [Graphics] sf::RenderStates ......   Passed    0.01 sec
      Start 21: [Graphics] sf::Shape
21/28 Test #21: [Graphics] sf::Shape .............   Passed    0.01 sec
      Start 22: [Graphics] sf::Transform
22/28 Test #22: [Graphics] sf::Transform .........   Passed    0.01 sec
      Start 23: [Graphics] sf::Transformable
23/28 Test #23: [Graphics] sf::Transformable .....   Passed    0.01 sec
      Start 24: [Graphics] sf::Vertex
24/28 Test #24: [Graphics] sf::Vertex ............   Passed    0.01 sec
      Start 25: [Graphics] sf::VertexArray
25/28 Test #25: [Graphics] sf::VertexArray .......   Passed    0.01 sec
      Start 26: [Graphics] sf::View
26/28 Test #26: [Graphics] sf::View ..............   Passed    0.01 sec
      Start 27: [Network] sf::IpAddress
27/28 Test #27: [Network] sf::IpAddress ..........   Passed    0.50 sec
      Start 28: [Network] sf::Packet
28/28 Test #28: [Network] sf::Packet .............   Passed    0.01 sec
```

---

I also have a commit that [simplifies how code coverage is applied](https://github.com/ChrisThrasher/SFML/commit/4c017a82e88dab855163b9d61fd05503a715ff0e) in case someone wants to check that out. I may add that this PR but chose to omit it for now so any discussion about it wouldn't delay reviewing the simpler stuff.

### CODE REVIEW

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Review by vittorioromeo — APPROVED** [review]
LGTM, nice cleanup!

File: test/TestUtilities/SystemUtil.hpp [hunk 1]
```diff
@@ -20,20 +20,20 @@ class Angle;
...
+std::ostream& operator<<(std::ostream& os, const String& string);
+std::ostream& operator<<(std::ostream& os, Time time);
 
 template <typename T>
-std::ostream& operator<<(std::ostream& os, const sf::Vector2<T>& vector)
+std::ostream& operator<<(std::ostream& os, const Vector2<T>& vector)
 {
     os << std::fixed << std::setprecision(std::numeric_limits<T>::max_digits10);
     os << "(" << vector.x << ", " << vector.y << ")";
     return os;
```
Comment (kimci86) [inline test/TestUtilities/SystemUtil.hpp:32]: While at it,
```suggestion
    return os << "(" << vector.x << ", " << vector.y << ")";
```

File: test/TestUtilities/SystemUtil.hpp [hunk 2]
```diff
@@ -20,20 +20,20 @@ class Angle;
...
 }
 
 template <typename T>
-std::ostream& operator<<(std::ostream& os, const sf::Vector3<T>& vector)
+std::ostream& operator<<(std::ostream& os, const Vector3<T>& vector)
 {
     os << "(" << vector.x << ", " << vector.y << ", " << vector.z << ")";
     return os;
```
Comment (kimci86) [inline test/TestUtilities/SystemUtil.hpp:39]:
```suggestion
    return os << "(" << vector.x << ", " << vector.y << ", " << vector.z << ")";
```

**Review by kimci86 — COMMENTED** [review]
Maybe `SFML/Config.hpp` test could be renamed to start with `[System]` also?

File: test/TestUtilities/SystemUtil.hpp (hunk 1 above)
Comment (ChrisThrasher) [reply test/TestUtilities/SystemUtil.hpp:32]: Thank you! I would have be so annoyed I missed that before this got merged.

**Comment by ChrisThrasher** [comment]
> Maybe `SFML/Config.hpp` test could be renamed to start with `[System]` also?

Yeah I suppose so. Any code which lives "outside" of any module is implicitly in the System module in some sense since that's the only mandatory module.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Review by kimci86 — APPROVED** [review]

**Pull request merged**